include/net/fota_download.h

```diff
@@ -42,11 +42,24 @@ enum fota_download_evt_id {
 	FOTA_DOWNLOAD_EVT_ERROR,
 };
 
+/**
+ * @brief FOTA download error cause values.
+ */
+enum fota_download_error_cause {
+	/** No error, used when event ID is not FOTA_DOWNLOAD_EVT_ERROR. */
+	FOTA_DOWNLOAD_ERROR_CAUSE_NO_ERROR,
+	/** Downloading the update failed. The download may be retried. */
+	FOTA_DOWNLOAD_ERROR_CAUSE_DOWNLOAD_FAILED,
+	/** The update is invalid and was rejected. Retry will not help. */
+	FOTA_DOWNLOAD_ERROR_CAUSE_INVALID_UPDATE,
+};
+
 /**
  * @brief FOTA download event data.
  */
 struct fota_download_evt {
 	enum fota_download_evt_id id;
+	enum fota_download_error_cause cause;
```

Comment:
I'd consider putting `cause` and `progress` into a union, these fields are releated to different events and won't be used at the same time.

Reply:
Yup, that's a good idea :+1:

Reply:
That's a good idea, I will do it. Thanks!